mypy/build.py

```diff
@@ -3078,7 +3078,7 @@ def load_graph(
                 path=bs.path,
                 source=bs.text,
                 manager=manager,
-                root_source=not bs.followed,
+                root_source=not (bs.followed or bs.module == "builtins"),
```

Comment:
Prior to this, `root_source` was True for `builtins`, which seemed incorrect? But I'm not sure of the intent.

The reason for adding this here is because of this check when constructing the `State`:

https://github.com/seddonym/mypy/blob/7e61402902f14fae93c0327d146172657f43c4c4/mypy/build.py#L1941